store/CSDK/test/main_ft.cpp

```diff
@@ -0,0 +1,1172 @@
+        writer.writtenBy("CSDK");
+        if (argc > 3) {
+            writer.withHadoopConf("fs.s3a.access.key", argv[1]);
+            writer.withHadoopConf("fs.s3a.secret.key", argv[2]);
+            writer.withHadoopConf("fs.s3a.endpoint", argv[3]);
+        }
+        writer.build();
+        int size = 10;
+        jclass objClass = env->FindClass("java/lang/String");
+
+        writeData(env, writer, size, objClass, "name3", 22);
+        writeData(env, writer, size, objClass, "name1", 11);
+        writeData(env, writer, size, objClass, "name2", 33);
+        writer.close();
+
+
+        carbonReader.builder(env, path);
+        carbonReader.build();
+        int i = 0;
+        CarbonRow carbonRow(env);
+        while (carbonReader.hasNext()) {
+            jobject row = carbonReader.readNextRow();
+            i++;
+            carbonRow.setCarbonRow(row);
+            printf("%d\t%s\t\n", carbonRow.getShort(0), carbonRow.getString(1));
+            env->DeleteLocalRef(row);
+        }
+        carbonReader.close();
+    } catch (jthrowable ex) {
+        env->ExceptionDescribe();
+        printf("%s",convertExceptionToString(ex));
+        env->ExceptionClear();
+        writer.close();
+        carbonReader.close();
+        return false;
+    }
+    return true;
+}
+
+/**
+ * read data from S3
+ * parameter is ak sk endpoint
+ *
+ * @param env jni env
+ * @param argv argument vector
+ * @return
+ */
+bool readFromS3(JNIEnv *env, char *path, char *argv[]) {
+    printf("\nRead data from S3:\n");
+    CarbonReader reader;
+
+    reader.builder(env, path, "test");
+    reader.withHadoopConf("fs.s3a.access.key", argv[1]);
+    reader.withHadoopConf("fs.s3a.secret.key", argv[2]);
+    reader.withHadoopConf("fs.s3a.endpoint", argv[3]);
+    reader.build();
+    printResult(env, reader);
+}
+
+
+
+TEST(CSDKTest,tryCatchException) {
+    try{
+        bool gotExp=tryCatchException(env);
+        EXPECT_TRUE(gotExp);
+    } catch (runtime_error e){
+        EXPECT_TRUE(true);
+    }
+}
+
+
+TEST(CSDKTest,tryCarbonRowException) {
+    char *smallFilePath = "../../../../resources/carbondata";
+    try {
+        bool result = tryCarbonRowException(env, smallFilePath);;
+        EXPECT_TRUE(result) << "Expected Exception as No Index File" << result;
+    } catch (runtime_error e) {
+        EXPECT_TRUE(true);
+    }
+}
+
+TEST(CSDKTest,testCarbonProperties) {
+    try {
+        bool result = testCarbonProperties(env);
+        EXPECT_TRUE(result) << "Carbon set properties not working ";
+    } catch (runtime_error e) {
+        EXPECT_TRUE(false) << " Exception is not expected while setting carbon properties";
+    }
+}
+
+TEST(CSDKTest,testWriteData) {
+    try {
+        bool result =testWriteData(env, "./data", my_argc, my_argv,10);
+        //incremental load
+        result = result && testWriteData(env, "./data", my_argc, my_argv,10);
+        EXPECT_TRUE(result) << "Either Data Loading Or Carbon Reader  is failed";
```

Comment:
Please take care the number of space, for example: 2 spaces between "Reader  is"